src/app/shared/components/header/header.component.ts

```diff
@@ -53,44 +55,42 @@ export class headerComponent implements OnInit, OnDestroy {
     private router: Router,
     private location: Location,
     private appConfigService: AppConfigService
-  ) {}
+  ) {
+    effect(
+      () => {
+        // when header config changes set the height and collapse properties
+        const { collapse, variant } = this.headerConfig();
+        this.setHeaderHeightCSS(variant);
+        if (collapse !== undefined) {
+          this.setHeaderCollapse(collapse);
+        }
+      },
+      { allowSignalWrites: true }
```

Comment:
My rough take on it is given how easy it is to accidentally end up in infinite loops it makes sense to be cautious about using `allowSignalWrites`. Pretty much any time you use there could be an alternate pattern available using `computed` signals instead (usually with additional intermediate variables), or by using regular variables and `cdr` methods to prompt change detection. 

I.e. in this example the signal write is so that the `marginTop` can be set following changes and automatically detected. If we just had a regular `marginTop` variable we could set and call `cdr.markForCheck`. I don't like this pattern as much as using both signals and cdr feels a bit wrong given they're meant to address the same problem

Alternatively you could make both `headerHeight` and `headerCollapse` computed properties that depend on headerConfig, and marginTop a computed property that depends on both. Again, I'm not a huge fan of this pattern, any case where there is one computed that depends on two others you can combine a single effect or duplicate some of the code and run as two separate functions which hide away the relationship between variables